src/core/ComplexFilter/index.tsx

```diff
@@ -126,7 +130,15 @@ export default function ComplexFilter<
 
   function handleClose() {
     if (multiple) {
-      setValue(pendingValue as Value<DefaultMenuSelectOption, Multiple>);
+      // Set pending value from ref on close. This needs to happen through a ref
```

Comment:
I just created another PR for the fix here 🙏 

https://github.com/chanzuckerberg/sci-components/pull/100

I think the closure is working as expected here, since when we select an option, React updates `pendingValue` as expected. (I added console.log('pendingValue', pendingValue) before the return statement to verify) 

And the reason why  `pendingValue` is `[]` by the time `handleClose()` is invoked is because `handleClick()` explicitly calls `setPendingValue(value)` BEFORE `handleClose` is invoked, so the solution in my PR is to remove that explicit call in `handleClick()`

PTAL thank you!